### PR TITLE
[GEOT-2425] Change from webdav to nexus repository

### PR DESCRIPTION
**Pull request by tbarsballe**
See https://github.com/geotools/geotools/wiki/Change-from-webdav-to-nexus-repository

Moved from repo.boundlessgeo.com to repo.osgeo.org.

## Checklist

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2317,16 +2300,16 @@
...
-      <id>osgeo</id>
-      <name>Open Source Geospatial Foundation - WebDAV upload</name>
-      <url>dav:http://download.osgeo.org/upload/geotools/</url>
     </repository>
     <snapshotRepository>
-        <id>boundless</id>
-        <uniqueVersion>false</uniqueVersion>
-        <name>Boundless Snapshot Repository</name>
-        <url>https://repo.boundlessgeo.com/snapshot/</url>
+      <id>nexus</id>
```
Comment (jodygarnett) [inline pom.xml:2309]: Does this id name need to be different from the release repo above? geotools-release and geotools-snapshot for example?

File: pom.xml [hunk 2]
```diff
@@ -2395,21 +2392,13 @@
...
     <pluginRepository>
       <snapshots>
         <enabled>true</enabled>
       </snapshots>
-      <id>boundless</id>
-      <name>Boundless Maven Repository</name>
-      <url>https://repo.boundlessgeo.com/snapshot/</url>
+      <id>osgeo</id>
+      <name>OSGeo Nexus Maven Repository</name>
+      <url>https://repo.osgeo.org/repository/snapshot/</url>
```
Comment (jodygarnett) [inline pom.xml:2401]: I have not setup any specific plugin repository, not sure if it is needed?

**Comment by jodygarnett** [comment]
It shows failures finding stuff:
```
[ERROR] Failed to execute goal on project gt-coverage: Could not resolve dependencies for project org.geotools:gt-coverage:jar:24-SNAPSHOT: The following artifacts could not be resolved: it.geosolutions.imageio-ext:imageio-ext-tiff:jar:1.3.2, ...
```

Which I can see in [the cache](https://repo.osgeo.org/#browse/browse:geo-solutions-cache:it%2Fgeosolutions%2Fimageio-ext%2Fimageio-ext-tiff%2F1.3.2)

Going to try and rerun ...

**Comment by aaime** [comment]
A note, if you want to try things out properly, you probably want to remove the maven repo cache usage. Right now it's building off a cache to reduce build times and avoid our builds from being blacklisted due to too many downloads (it has happened).

You can remove it to do the tests:
https://github.com/geotools/geotools/blob/master/.travis.yml#L3
and then set it back once it's all green.

**Comment by jodygarnett** [comment]
Travis is not enjoying this experience:

```
[INFO] Installing /home/travis/build/geotools/geotools/modules/unsupported/arcsde/common/target/gt-arcsde-common-24-SNAPSHOT-sources.jar to /home/travis/.m2/repository/org/geotools/gt-arcsde-common/24-SNAPSHOT/gt-arcsde-common-24-SNAPSHOT-sources.jar
3541
3542
3543No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
3544Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

@tbarsballe I will check out and build locally I think this may be too fundamental a change for our build bots.

**Comment by tbarsballe** [comment]
Sounds good. I'll restart the travis builds to see if that particular failure was a one-time thing as I notice the QA build ran fine, as did the windows one.

**Comment by jodygarnett** [comment]
Completed `mvn clean install -Dall` locally on fresh repository (Java 1.8.0_211):
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14:36 min
[INFO] Finished at: 2020-04-07T10:10:47-07:00
[INFO] ------------------------------------------------------------------------
```

**Comment by jodygarnett** [comment]
So what is needed next? Do we merge this, do a similar change for geowebcache and geoserver?

**Comment by tbarsballe** [comment]
Alright, shall we just revert the Travis.yml change and merge this then, and sort things out on Jenkins as they occur?

My only real concern is whether this will cause other travis builds to fail once its merged...

**Comment by jodygarnett** [comment]
Starting a branch for geowebcache, will ask you to review.
Then I think we merge them in order, and clean up the mess.
And then backport to active branches.

**Comment by tbarsballe** [comment]
I think we should merge this, kick off a build on Jenkins to populate the snapshots repo, then continue with GWC and GeoServer. Do you want to take a look at GWC? I don't have a checkout of it right now.

File: pom.xml (hunk 1 above)
Comment (jodygarnett) [reply pom.xml:2309]: @tbarsballe I would like to use consistent naming for these repositories to make the documentation easier to follow.

Can we make the change now?

**Comment by tbarsballe** [comment]
Sounds good to me. I'll revert the travis change, merge this, and kick off GT builds on Jenkins

**Comment by tbarsballe** [comment]
As noted in the GeoServer PR, having different names for the distribution repositories makes managing credentials in settings.xml a pain (download repo names don't matter, since they can connect unauthenticated), so I'd rather not do that. Having all the distribution repositories named "nexus" works fine, and makes this much easier to work with

File: pom.xml [hunk 3]
```diff
@@ -2365,21 +2338,45 @@
     </repository>
 
     <repository>
-      <!-- Contains snapshot and release (including third-party-dependencies)              -->
+      <id>osgeo-releases</id>
+      <name>OSGeo Nexus Release Repository</name>
+      <url>https://repo.osgeo.org/repository/release/</url>
+      <!-- contains release (including third-party-dependences)                            -->
       <!-- Restlet maven Repository (http://maven.restlet.org)                             -->
```
Comment (jodygarnett) [inline pom.xml:2345]: This repo does not cache restlet.org so this line could be removed.

Comment (jodygarnett) [reply pom.xml:2345]: I ended up with:
```
      <!-- Contains releases for OSGeo projects: GeoTools, GeoWebCache, GeoServer, ...     -->
      <!-- Includs third-party-dependences, and caches of:                                 -->
      <!-- ucar (https://artifacts.unidata.ucar.edu/content/repositories/unidata-releases) -->
      <!-- geosolutions (http://maven.geo-solutions.it/)                                   -->
```

File: pom.xml [hunk 4]
```diff
@@ -2317,16 +2300,16 @@
...
-      <name>Open Source Geospatial Foundation - WebDAV upload</name>
-      <url>dav:http://download.osgeo.org/upload/geotools/</url>
     </repository>
     <snapshotRepository>
-        <id>boundless</id>
-        <uniqueVersion>false</uniqueVersion>
-        <name>Boundless Snapshot Repository</name>
-        <url>https://repo.boundlessgeo.com/snapshot/</url>
+      <id>nexus</id>
+      <uniqueVersion>false</uniqueVersion>
```
Comment (jodygarnett) [inline pom.xml:2310]: Apparently unqiueversion is ignored for maven 3 and this line can be removed.

File: pom.xml (hunk 3 above)
Comment (tbarsballe) [reply pom.xml:2345]: From https://github.com/geoserver/geoserver/pull/4165 , we ended up having to mirror restlet, so leaving this unchanged, as it is now accurate, and the repo does cache restlet.

File: pom.xml (hunk 4 above)
Comment (tbarsballe) [reply pom.xml:2310]: On that note, I noticed that all our nexus repos are "maven2" repositories. Do you know if that would effect anything, even if the build is done with a maven 3 release?

**Pull request merged**